src/main/kotlin/org/rust/lang/core/lexer/RustStringLiteralLexer.kt

```diff
@@ -0,0 +1,137 @@
+package org.rust.lang.core.lexer
+
+import com.intellij.openapi.util.text.StringUtil
+import com.intellij.psi.StringEscapesTokenTypes.*
+import com.intellij.psi.tree.IElementType
+
+private const val BYTE_ESCAPE_LENGTH = "\\x00".length
+private const val UNICODE_ESCAPE_MIN_LENGTH = "\\u{0}".length
+private const val UNICODE_ESCAPE_MAX_LENGTH = "\\u{000000}".length
+
+// TODO: Maybe implement lexing of prefixes and suffixes?
```

Comment:
Ah yeah, suffixes are specified for forwards compatibility only, I think we should not pay attention to them.

As for prefixes, I think that the prefix should be uniquely identified by literal type?